src/credentials/tests/TestGroupDataProvider.cpp

```diff
@@ -1164,126 +1216,36 @@ void TestGroupDecryption(nlTestSuite * apSuite, void * apContext)
     auto it      = provider->IterateGroupSessions(session_id);
     size_t count = 0, total = 0;
 
-    NL_TEST_ASSERT(apSuite, it);
-    if (it)
+    EXPECT_TRUE(it);
```

Comment:
it seems the previous code did the equivalent of ASSERT_TRUE here (i.e. early exit).

```suggestion
    ASSERT_TRUE(it);
```

Reply:
Fixed